apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -111,39 +116,40 @@ internal_resolve(Protocol, DNSMessage) ->
     Response = erldns_handler:do_handle(DNSMessage, ?LOCALHOST),
     encode_message(Protocol, Response).
 
--spec(resolve(protocol(), [upstream()], binary(), reply_fun()) -> ok).
-resolve(Protocol, Upstreams, Request, Fun) ->
+-spec(internal_resolve(protocol(), [upstream()], binary(), reply_fun(), binary())-> ok).
+internal_resolve(Protocol, Upstreams, Request, Fun, Zone) ->
```

Comment:
I felt that having `resolve/3` in the exposed API whit `internal_resolve/3` and `resolve/4` broke my expectations. I renamed `resolve/4` to `internal_resolve/4` to solve that for me. Maybe you will also find that an improvement :)